EventListener/ViewResponseListener.php

```diff
@@ -87,14 +87,10 @@ public function onKernelView(GetResponseForControllerResultEvent $event)
                 $view->setStatusCode($configuration->getStatusCode());
             }
             if ($configuration->getSerializerGroups() && !$customViewDefined) {
-                $context = $view->getSerializationContext() ?: new SerializationContext();
-                $context->setGroups($configuration->getSerializerGroups());
-                $view->setSerializationContext($context);
+                $context->addGroups($configuration->getSerializerGroups());
             }
             if ($configuration->getSerializerEnableMaxDepthChecks()) {
-                $context = $view->getSerializationContext() ?: new SerializationContext();
-                $context->enableMaxDepthChecks();
-                $view->setSerializationContext($context);
+                $context->setMaxDepth(0);
```

Comment:
is this really equivalent? ``enableMaxDepthChecks()`` just enables checking for max depth, where the max depth is then actually "dynamic", ie. defined on the relevant mapping data in the relevant entities